document-generation/smartdocuments/src/main/kotlin/com/ritense/smartdocuments/client/SmartDocumentsClient.kt

```diff
@@ -43,14 +49,33 @@ import org.springframework.web.reactive.function.client.ExchangeFilterFunctions
 import org.springframework.web.reactive.function.client.ExchangeStrategies
 import org.springframework.web.reactive.function.client.WebClient
 import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
 
 class SmartDocumentsClient(
     private var smartDocumentsConnectorProperties: SmartDocumentsConnectorProperties,
     private val smartDocumentsWebClientBuilder: WebClient.Builder,
     private val maxFileSizeMb: Int,
     private val temporaryResourceStorageService: TemporaryResourceStorageService,
+    private val pluginService: PluginService
 ) {
 
+    fun getDocumentStructure(): DocumentsStructure? {
+        return pluginWebClient().get()
+                .uri(STRUCTURE_PATH)
+                .retrieve()
+                .onStatus(
+                    { statusCode -> statusCode.is4xxClientError || statusCode.is5xxServerError },
+                    { clientResponse ->
+                        val statusCode = clientResponse.statusCode()
+                        logger.warn { "SmartDocuments has not returned a DocumentStructure. Status code: $statusCode" }
+                        Mono.empty()
```

Comment:
The client must throw an error when something goes wrong. You can decide to catch this error in the plugin-action if you think its not relevant.